app/grandchallenge/components/serializers.py

```diff
@@ -26,32 +30,130 @@ class SimpleImageSerializer(serializers.ModelSerializer):
     # name is needed
     class Meta:
         model = Image
-        fields = (
-            "pk",
-            "name",
-        )
+        fields = ("pk", "name")
 
 
-class ComponentInterfaceValueSerializer(serializers.ModelSerializer):
-    # Serializes images in place rather than with hyperlinks for internal usage
-    image = SimpleImageSerializer()
-    interface = ComponentInterfaceSerializer()
+class ComponentInterfaceValuePostSerializer(serializers.ModelSerializer):
+    """
+    Serializes images with hyperlinks for external usage
+    Expects interface_slug, to allow creating a ComponentInterfaceValue with an
+    existing ComponentInterface
+    """
+
+    image = serializers.HyperlinkedRelatedField(
+        queryset=Image.objects.all(),
+        view_name="api:image-detail",
+        required=False,
+    )
+
+    interface_slug = serializers.CharField(write_only=True)
+    interface = ComponentInterfaceSerializer(read_only=True)
+    upload_pk = serializers.CharField(write_only=True, required=False)
```

Comment:
Possibly [PrimaryKeyRelatedField](https://www.django-rest-framework.org/api-guide/relations/#primarykeyrelatedfield) can be used here to make it more strict. This would also allow you to pass a queryset and filter on session.status.

Reply:
I opted for the HyperLinkedRelatedField to be more in line with the image (also HyperLinkedRelatedField)